packages/clay-date-picker/src/DateNavigation.tsx

```diff
@@ -0,0 +1,150 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as Helpers from './Helpers';
+import moment from 'moment';
+import React, {
+	ChangeEvent,
+	FunctionComponent,
+	HTMLAttributes,
+	useMemo,
+} from 'react';
+import {IAriaLabels, IYears} from './ClayDatePicker';
+
+import Button from './Button';
+import Select, {ISelectOption} from './Select';
+
+interface Props extends HTMLAttributes<HTMLDivElement> {
+	ariaLabels: IAriaLabels;
+	currentMonth: Date;
+	months: Array<string>;
+	onDotClicked: () => void;
+	onMonthChange: (date: Date) => void;
+	spritemap: string;
+	years: IYears;
+}
+
+const DateNavigation: FunctionComponent<Props> = ({
+	ariaLabels,
+	currentMonth,
+	months,
+	onDotClicked,
+	onMonthChange,
+	spritemap,
+	years,
+}) => {
+	const memoizedYears: Array<ISelectOption> = useMemo(
+		() =>
+			Helpers.range(years).map(elem => {
+				return {
+					label: elem,
+					value: elem,
+				};
+			}),
+		[years]
+	);
+
+	const memoizedMonths: Array<ISelectOption> = useMemo(
+		() =>
+			months.map((month, index) => {
+				return {
+					label: month,
+					value: index,
+				};
+			}),
+		[months]
+	);
+
+	/**
+	 * Handles the change of the month from the available
+	 * years in the range
+	 */
+	function handleChangeMonth(month: number) {
+		const date = moment(currentMonth)
+			.clone()
+			.add(month, 'M')
+			.toDate();
+		const year = date.getFullYear();
+
+		if (memoizedYears.find(elem => elem.value === year)) {
+			onMonthChange(date);
+		}
+	}
+
+	/**
+	 * Handles the previous month button
+	 */
+	const handlePreviousMonthClicked = () => handleChangeMonth(-1);
+
+	/**
+	 * Handles the next month button
+	 */
+	const handleNextMonthClicked = () => handleChangeMonth(1);
+
+	/**
+	 * Handles the change of the year and month of the header
+	 */
+	function handleFormChange(event: ChangeEvent<HTMLSelectElement>) {
+		if (event.currentTarget.form === null) return;
```

Comment:
Kind of surprised that we don't have an eslint rule that prohibits conditionals without braces.

Reply:
I think there is a CSF rule against it, which means we should probably add it to our `eslint-config-liferay`